homeassistant/components/notify/__init__.py

```diff
@@ -60,17 +60,53 @@
 @bind_hass
 async def async_reload(hass, integration_name):
     """Register notify services for an integration."""
+    await _do_with_notify_services(hass, integration_name, _async_setup_notify_services)
+
+
+@bind_hass
+async def async_reset_platform(hass, integration_name):
+    """Unregister notify services for an integration."""
+    if not await _do_with_notify_services(
+        hass, integration_name, _async_unregister_notify_services
+    ):
+        return
+    del hass.data[NOTIFY_SERVICES][integration_name]
+
+
+async def _do_with_notify_services(hass, integration_name, call):
+    """Call a method on all notify services for an integration."""
     if (
         NOTIFY_SERVICES not in hass.data
         or integration_name not in hass.data[NOTIFY_SERVICES]
     ):
-        return
+        return False
 
-    tasks = [
-        _async_setup_notify_services(hass, data)
-        for data in hass.data[NOTIFY_SERVICES][integration_name]
-    ]
+    tasks = [call(hass, data) for data in hass.data[NOTIFY_SERVICES][integration_name]]
     await asyncio.gather(*tasks)
+    return True
+
+
+async def _async_unregister_notify_services(hass, data):
+    """Remove the notify services."""
+    targets = data[TARGETS]
+
+    if targets:
+        remove_targets = set(targets)
+        for remove_target_name in remove_targets:
+            del targets[remove_target_name]
+            hass.services.async_remove(
+                DOMAIN,
+                remove_target_name,
+            )
+
+    friendly_name_slug = slugify(data[FRIENDLY_NAME])
```

Comment:
This works. In a future PR we should move all service registration things to `BaseNotificationService` and store the name of the service we're registered as.

Reply:
Hmm maybe we should do it in this PR actually.

Reply:
I'm trying to think of a good way to make that work with all the integrations providing their own `__init__` and never calling `super().__init__`

Reply:
I ended up creating an `async_setup` in the `BaseNotificationService`.  I wanted to put it in `__init__` but that would break all the existing.  It seemed cleaner than having an external data class

Reply:
I worked around the lack of `super().__init__` being called in a cleaner way.

Reply:
Well every integration seems to have defined their own init, and some are re-passing in `hass` so I had to go back to doing the `async_setup`.